daemon/volumes.go

```diff
@@ -257,7 +258,11 @@ func (daemon *Daemon) registerMountPoints(container *container.Container, hostCo
 				StorageOpt: container.HostConfig.StorageOpt,
 			}
 
-			layerName := fmt.Sprintf("%s-%s", container.ID, mp.Source)
+			// Include the destination in the layer name to make it unique for each mount point and container.
+			// This makes sure that the same image can be mounted multiple times with different destinations.
+			// Hash the destination to create a safe, unique identifier
+			destHash := sha256.Sum256([]byte(container.ID + "src=" + mp.Source + "dst=" + mp.Destination))
+			layerName := hex.EncodeToString(destHash[:])
```

Comment:
Would we need to correlate the ID with these anywhere? Considering if (e.g.) base64 encoding or something like that would work instead of a cryptographic hash.

What are the constraints on these names? (is this path on disk, or otherwise?)

Reply:
We don't really need to correlate these, we just need "a" name for the snapshot. I think it could only be problematic when trying to match the name when filtering snapshots?

Also not sure if there's a length limit on snapshot ids.

Reply:
Changed to hex-encoding, so it's reversible in case it's needed.